### PR TITLE
Feature contribution

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

File: src/app/components/demo-component.service.ts [hunk 1]
```diff
@@ -156,17 +53,96 @@ export class StacheDemoComponentService {
...
+    //   route: '/contribute/components/layout-container',
+    //   icon: 'map-o',
+    //   summary: 'Basic page layouts using Stache and skyux pattern, for quick page layouts.'
+    // },
+    // {
+    //   name: 'Layout: Sidebar',
+    //   route: '/contribute/components/layout-sidebar',
+    //   icon: 'map-o',
+    //   summary: 'Basic page layouts using Stache and skyux pattern, for quick page layouts.'
+    // },
```
Comment (Blackbaud-BrandonJones) [inline src/app/components/demo-component.service.ts]: can we remove this commented code?    I think your suggestion to remove it and add it into an issue log will keep this a lot cleaner.

File: src/app/contribute/components/back-to-top/index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    </ul>
+  </stache-page-summary>
+
+  <stache-page-anchor>
+    Demo
+  </stache-page-anchor>
+  <div class="stache-demo-page-result">
+    <stache-back-to-top offset="1"></stache-back-to-top>
+    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Asperiores earum veniam incidunt rerum porro tenetur possimus natus eligendi, officia quasi est voluptatum, ab distinctio consequatur minus, nesciunt optio qui! Voluptate?</p>
+    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Asperiores earum veniam incidunt rerum porro tenetur possimus natus eligendi, officia quasi est voluptatum, ab distinctio consequatur minus, nesciunt optio qui! Voluptate?</p>
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/back-to-top/index.html]: Spacing with content inside of a non-void tag.
```
<p>
       Lorem ....
</p>
```

File: src/app/contribute/components/contribute-components.component.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<stache
+  pageTitle="Development Components"
+  layout="container">
+
+  <p>These components are Angular 2 components, services, and modules that serve primarily to facilitate the abstractions built in to Stache. They can be used independently, but are not a public part of the Stache content creation library.</p>
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/contribute-components.component.html]: Same at the spacing on the <p> tags above.

File: src/app/contribute/components/contribute-components.component.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  templateUrl: './contribute-components.component.html'
+})
+export class StacheContributeComponentsComponent implements OnInit {
+  public components: StacheDemoComponent[];
+  public routes: StacheNavLink[] = [];
+
+  public constructor(private componentService: StacheDemoComponentService) { }
+
+  public ngOnInit(): void {
+    this.components = this.componentService.getAllDevelopmentSorted();
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/contribute-components.component.ts]: do we need this public variable?  Or can we make it a local scoped variable instead?

File: src/app/contribute/components/page-header/index.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Page Header"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/page-header/index.html]: sidebar is now the default.

File: src/app/contribute/components/page-header/index.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  showBreadcrumbs="true">
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/page-header/index.html]: breadcrumbs also default to true.

File: src/app/contribute/components/table-of-contents/index.html [hunk 7]
```diff
@@ -0,0 +1,61 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Table of Contents"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/table-of-contents/index.html]: same as above regarding defaults.

File: src/app/components/demo-component.service.ts (hunk 1 above)
Comment (blackbaud-brandonhare) [inline src/app/components/demo-component.service.ts]: Done

File: src/app/contribute/components/back-to-top/index.html [hunk 8]
```diff
@@ -0,0 +1,44 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Back To Top"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/back-to-top/index.html]: defaults.

File: src/app/contribute/components/breadcrumbs/index.html [hunk 9]
```diff
@@ -0,0 +1,60 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Breadcrumbs"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/breadcrumbs/index.html]: defaults

File: src/app/contribute/components/container/index.html [hunk 10]
```diff
@@ -0,0 +1,33 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Container"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/container/index.html]: defaults

File: src/app/contribute/components/nav/index.html [hunk 11]
```diff
@@ -0,0 +1,67 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Nav"
+  layout="sidebar"
+  showBreadcrumbs="true">
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/nav/index.html]: defaults

File: src/app/contribute/components/page-title/index.html [hunk 12]
```diff
@@ -0,0 +1,33 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Page Header"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/page-title/index.html]: defaults

File: src/app/contribute/components/sidebar/index.html [hunk 13]
```diff
@@ -0,0 +1,98 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Sidebar"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/components/sidebar/index.html]: defaults

File: src/app/contribute/index.html [hunk 14]
```diff
@@ -0,0 +1,10 @@
+<app-nav></app-nav>
+<stache
+  pageTitle="Contribute"
+  layout="sidebar"
```
Comment (Blackbaud-BrandonJones) [inline src/app/contribute/index.html]: defaults

**Pull request merged**